cli/src/commands/parse.rs

```diff
@@ -76,63 +58,37 @@ pub fn try_get_packages(path: &Path) -> Result<(Vec<PackageDescriptor>, PackageT
 /// Determine the lockfile type based on its name and parse
 /// accordingly to obtain the packages from it
 pub fn get_packages_from_lockfile(path: &Path) -> Result<(Vec<PackageDescriptor>, PackageType)> {
-    let file = path
-        .file_name()
-        .and_then(|file| file.to_str())
-        .ok_or_else(|| anyhow!("Lockfile path has no file name"))?;
-    let ext = path.extension().and_then(|ext| ext.to_str());
-
-    let pattern = match ext {
-        Some("csproj") => ".csproj",
-        _ => file,
-    };
-
-    let res = match pattern {
-        "Gemfile.lock" => parse(GemLock, path)?,
-        "package-lock.json" => parse(PackageLock, path)?,
-        "yarn.lock" => parse(YarnLock, path)?,
-        "requirements.txt" => parse(PyRequirements, path)?,
-        "Pipfile" | "Pipfile.lock" => parse(PipFile, path)?,
-        "poetry.lock" => parse(Poetry, path)?,
-        "effective-pom.xml" => parse(Pom, path)?,
-        "gradle.lockfile" => parse(GradleLock, path)?,
-        ".csproj" => parse(CSProj, path)?,
-        _ => try_get_packages(path)?,
+    let res = if let Some(parser) = get_path_parser(path) {
+        let data = read_to_string(path)?;
+        (parser.parse(&data)?, parser.package_type())
+    } else {
+        try_get_packages(path)?
```

Comment:
I'd say with option it's more consistent internally and generally within the Rust ecosystem to `match` rather than using `if let/else`.